1.10.2 loading screen

okushama opened the pull request.

okushama committed:
added loading screen

screen subtitles populate via categories from the mod language file
renders two mobs from the tropic raft egg list (minus some irregularly
sized ones)
have tweaked water mob texture selection code to account for this

okushama committed:
pre-emptive strike

okushama committed:
wrapped in GUI for extra sadness

:c

okushama committed:
.

okushama committed:
.

okushama committed:
.

okushama committed:
says farewell when you leave

okushama committed:
added support for other mods

to not have a farewell screen on their dimension entry from ours,
TF plans to add the same, so i fire the message in the hopes they
receive it and disable their farewell for us

added categories for mobs, only relevant ones to the background will be
shown

okushama committed:
randomised items from a list for display

okushama committed:
>_>

okushama committed:
dem too

seriously though, fight me

okushama committed:
*gives up*

okushama committed:
removed call from initGui causing reloads on resize

okushama committed:
>__>; left in debug

okushama committed:
removed kept instance of gui

randomised animated entity startup rotation

okushama committed:
Merge branch '1.10.2' into 1.10.2_loading_screen

okushama committed:
didn't add anything, don't know what you're on about

Cojomax99 commented on src/main/java/net/tropicraft/client/gui/GuiTropicsLoading.java:
```diff
@@ -0,0 +1,312 @@
+	private static final String[] MOBS_WATER = { "marlin", "dolphin", "hammerhead", "turtle", "seahorse", "failgull" };
+	private static final String[] MOBS_LAND = { "iguana", "failgull", "redfrog", "bluefrog", "greenfrog", "yellowfrog",
+			"monkey" };
+	private static final String[] MOBS_VILLAGE = { "koa", "ashen", "tropicreeper", "tropiskelly" };
```
I see these becoming problematic in the future if we change one of the registry names. This doesn't need to be done now, but we should probably move those registry names to an enum or at least string constants that can be referenced anywhere. Just a task worthy of going on the backlog for now.

tterrag1098 commented on src/main/java/net/tropicraft/client/gui/GuiTropicsLoading.java:
```diff
@@ -0,0 +1,312 @@
+
+	@Override
+	public void updateScreen() {
+		++this.progress;
```
Perhaps this should extend GuiDownloadTerrain to avoid duplicating this logic?

okushama committed:
implemented PR changes

Loading:
- changed from entity string to class references for a bit more
future-proofing
- now extends GuiDownloadTerrain

Scuba:
- cancelled motion override if the player is detected as flying
